vignettes/epiparameter.Rmd

```diff
@@ -0,0 +1,459 @@
+
+```{r, epinow2 with epiparameter, eval=FALSE}
+incubation_period <- epidist(
+  pathogen = "SARS_CoV_2_wildtype", 
+  delay_dist = "incubation"
+)
+serial_interval <- epidist(
+  pathogen = "SARS_CoV_2_wildtype", 
+  delay_dist = "serial_interval"
+)
+
+reporting_delay <- list(
+  mean = convert_to_logmean(3, 1), 
+  mean_sd = 0.1,
+  sd = convert_to_logsd(3, 1), 
+  sd_sd = 0.1, 
+  max = 10
+)
+
+reported_cases <- example_confirmed[1:40] 
+
+# use lnorm_logmeansd2meansd() to convert the parameters of the lognormal to the
+# mean (expectation) and standard deviation
+serial_interval_mean_sd <- lnorm_logmeansd2meansd(
+  meanlog = serial_interval$param[["meanlog"]], 
+  sdlog = serial_interval$param[["sdlog"]]
+)
+
+incubation_period_mean_sd <- lnorm_logmeansd2meansd(
+  meanlog = incubation_period$param[["meanlog"]], 
+  sdlog = incubation_period$param[["sdlog"]]
+)
+
+# for now serial interval is used in place of generation time
+generation_time_list <- list(
+  mean = serial_interval_mean_sd$mean
+  mean_sd = 0.1,
+  sd = serial_interval_mean_sd$sd,
+  sd_sd = 0.1,
+  max = 20
+) 
+
+incubation_period_list <- list(
+  mean = incubation_period_mean_sd$mean,
+  mean_sd = 0.1,
+  sd = incubation_period_mean_sd$sd,
+  sd_sd = 0.1,
+  max = 20
+)
+
+# estimate Rt by date of infection
+out <- epinow(
+  reported_cases = reported_cases, 
+  generation_time = generation_time_list,
+  rt = rt_opts(prior = list(mean = 2, sd = 0.1)),
+  delays = delay_opts(incubation_period_list, reporting_delay),
+  stan = stan_opts(samples = 750, chains = 4),
+  return_output = TRUE,
+  verbose = FALSE
+)
+```
+
+### Estimating $R_t$ using `R0`
+
+The R package `R0` contains a set of functions to estimate the basic reproduction number ($R_0$) and the effective reproduction number ($R_t$). Here we demonstrate a calculate of the time-varying $R_t$ using the method of @wallinga_different_2004. This method uses the epidemic curve of incidence data over the course of an epidemic and the generation time. Here we calculate $R_t$ using data on the outbreak of ebola from Sierra Leone from 2014 from `outbreaks`, and using the a serial interval from `epiparameter` as a proxy for the generation time.
+
+```{r R0 with epiparameter}
+data("ebola_sierraleone_2014")
+
+ebola_incidence <- incidence(
+  x = ebola_sierraleone_2014, 
+  date_index = "date_of_onset"
+)
+
+# format incidence data for `estimate.R()`
+ebola_incidence_R0 <- ebola_incidence$count
+names(ebola_incidence_R0) <- ebola_incidence$date_index
+
+ebola_si <- epidist(pathogen = "ebola", delay_dist = "serial_interval")
+
+si_mean_sd <- gamma_shapescale2meansd(
+  shape = ebola_si$param[["shape"]], 
+  scale = ebola_si$param[["scale"]]
+)
+
+mGT <- generation.time(type = "gamma", val = c(si_mean_sd$mean, si_mean_sd$sd))
```

Comment:
Would it be worth mentioning we take the serial time as a proxy for the generation time, which is not exact but probably good enough?
Maybe something to confirm with Adam.